app/components/templates/templatesController.js

```diff
@@ -46,6 +46,35 @@ function ($scope, $q, $state, $transition$, $anchorScroll, $filter, ContainerSer
     return true;
   }
 
+  $scope.pullImage = function() {
+    $('#createContainerSpinner').show();
+
+    var userDetails = Authentication.getUserDetails();
+    var accessControlData = $scope.formValues.AccessControlData;
+    var isAdmin = userDetails.role === 1 ? true : false;
+
+    if (!validateForm(accessControlData, isAdmin)) {
+      $('#createContainerSpinner').hide();
+      return;
+    }
```

Comment:
You can move `$('#createContainerSpinner').show()` function after this condition and therefore reduce branches content.